netweaver/ha_cluster.sls

```diff
@@ -81,6 +81,16 @@ adapt_sap_profile_ascs_{{ instance_name }}:
     - pattern: '^Restart_Program_01 = local \$\(_EN\) pf=\$\(_PF\)'
     - repl: 'Start_Program_01 = local $(_EN) pf=$(_PF)'
 
+set_keepalive_option_{{ instance_name }}:
+  file.line:
+    - name: {{ profile_file }}
+    - mode: insert
+    - location: end
+    - content: enque/encni/set_so_keepalive = true
+    # onlyif statements can be improved when salt version 3000 is used
+    # https://docs.saltstack.com/en/latest/ref/states/requisites.html#onlyif
+    - onlyif: cat /etc/salt/grains | grep "ensa_version_{{ node.sid.lower() }}_{{ instance }}:.*1"
```

Comment:
We can get version from pillar as you used in other file, but this seems good too.
And does the line in content: `enque/encni/set_so_keepalive = true` needs qoutes?

Reply:
No, we don't need quotes. At least following the best practices guide.